### DISCUSSION TITLE
Service characteristics changed event

### DISCUSSION THREAD

**Discussion by sanastasiou** [capacitor-community/bluetooth-le#558] [Q&A]
Hi all,

does anyone know if the following is supported by this plugin ?

https://developer.apple.com/forums/thread/120689

Basically I have a BLE device which depending on some external input would add/remove certain characteristics from a service.

So far, I was unable to see the new characteristics via a new scan / disconnect / reconnect although while using Light Blue App I can verify that the new characteristics were added / removed successfully as expected. 

Is this a bug of this plugin or some not implemented feature? 

If this is a known issue, is there any workaround to flush the cached service properties and force a re-read ?

Best Regards

**Comment by pwespi** [comment]
So even when you call [discoverServices](https://github.com/capacitor-community/bluetooth-le#discoverservices) or disconnect + reconnect, the new characteristic does not show up?

**Comment by sanastasiou** [comment]
@pwespi I did not try yet discoverServices, I did definitely disconnected and reconnected though. I will try also using discoverServices and let you know again. Thanks for the tip!

**Comment by sanastasiou** [comment]
@pwespi plugin version 1.9.2 does not offer an API `discoverServices`. Is this the same as `getServices`?

**Comment by sanastasiou** [comment]
Oh I see this was added only in the latest and greatest version of the plugin.. So time to upgrade it seems. I will get back to you!

**Comment by sanastasiou** [comment]
I upgraded my project to max ionic version.. now the phone is out of date. New phone arrives today. Will retest and let u know!

**Comment by sanastasiou** [comment]
So, I just tested this with a Pixel 4, Android 13. 

Same issue. Issue persists even after reconnecting and even after manually calling discoverServices.

If I restart the app, once the peripheral has started transmitting the additional characteristics, it works fine, but if the peripheral changes the service in any way on the fly it appears to be impossible to update the app cached service without restarting it.

**Comment by sanastasiou** [comment]
@pwespi is it possible that we are dealing with some issue with the plugin here? Since LightBlue on the same phone does detect the changes?

**Reply by pwespi** [reply]
Unfortunately I cannot reproduce your problem. When I add or remove a characteristic on my test device and then call `discoverServices` and `getServices`, I see the update in my app even without disconnecting.

**Reply by sanastasiou** [reply]
That's good to know.. perhaps I am doing something wrong???.. I am gonna try again and let u know! Thanks for the feedback!

**Reply by sanastasiou** [reply]
What kind of test device are u using btw?

**Reply by pwespi** [reply]
For this I used a second phone with the nRF connect app.

**Reply by sanastasiou** [reply]
Got it. Does your change ( i.e. addition of characteristic ) also include changing the advertisement ? I am guessing that my error might lie there.

**Comment by sanastasiou** [comment]
So this works indeed fine. The issue was following: I had paired the devices before and after that I did some modifications to the Service being offered. When that happens one needs to delete/repair the bond otherwise nothing works properly. 

Once that is done, everything works perfectly. Discover Services call is not even required.

Thanks!
